src/test/java/com/worksap/fig/HashTest.java

```diff
@@ -59,4 +59,31 @@ public void testFilter(){
         filtered.put(1, 2);
         assertTrue(filtered.equals(hash.filter((k, v) -> k + v != 6)));
     }
+
+    @Test(expected = NullPointerException.class)
+    public void testReject$() {
+        Hash<Integer, Integer> hash = Hash.newHash();
+        hash.put(1, 2);
+        hash.put(2, 4);
+        hash.put(5, 1);
+        assertEquals(2, hash.reject$((k, v) -> k > 1 && v > 1).size());
+        assertEquals(null, hash.get(2));
+        assertEquals(2, hash.reject$((k, v) -> k == v).size());
+        assertEquals(Hash.newHash(), hash.reject$((k, v) -> k != v));
+        hash.reject$(null);
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void testFilter$() {
+        Hash<Integer, Integer> hash = Hash.newHash();
+        hash.put(1, 2);
+        hash.put(2, 4);
+        hash.put(5, 1);
+        assertEquals(2, hash.filter$((k, v) -> !(k > 1 && v > 1)).size());
+        assertEquals(null, hash.get(2));
+        assertEquals(2, hash.filter$((k, v) -> k != v).size());
+        assertEquals(Hash.newHash(), hash.filter$((k, v) -> k == v));
+        hash.filter$(null);
+    }
+
 }
```

Comment:
Could you modify test case for `testReject` `testFilter` (not the two you committed), to ensure they don't change the original object?